### PR TITLE
Soycode fixtests

### PR DESCRIPTION
**Pull request by soycode**
Meant to fix the failing TCP test (there also seems to be a spotty XHR test, but it passed last run at least...).

### CODE REVIEW

File: providers/core.tcpsocket.js [hunk 1]
```diff
@@ -317,7 +330,8 @@ TcpSocket_node.prototype.listen = function (address, port, callback) {
 
   this.connection = this.net.createServer();
   this.callback = callback;
-  this.state = TcpSocket_node.state.BINDING;
+  TcpSocket_node.connectionState[this.id] = this.state =
```
Comment (willscott) [inline providers/core.tcpsocket.js]: these should all be 2 lines.

**Comment by soycode** [comment]
Not sure why tests are failing in Shippable - they're passing in Travis and locally (Linux VM).

File: providers/core.tcpsocket.js (hunk 1 above)
Comment (soycode) [inline providers/core.tcpsocket.js]: Done.

**Comment by soycode** [comment]
Shippable is just super flaky, if I kick the test repeatedly it sometimes passes (e.g. https://app.shippable.com/builds/55b122a8be4ac90b00eaec14) and sometimes doesn't. It passes reliably locally on a Linux VM which I would hope is comparable to Shippable, but there must be some subtle difference about how it's handling some of the networking/sockets.

**Comment by soycode** [comment]
Looks to pass more reliably on node 0.10 than 0.11. Hopefully some change between the two explains/allows us to fix.

**Comment by soycode** [comment]
@willscott @ryscheng LMK if any other thoughts/review, I plan to merge this and #34 sometime this week to cut a fresh release.

**Pull request merged**